src/transport/PeerConnections.h

```diff
@@ -236,6 +237,51 @@ class PeerConnections
         return state;
     }
 
+    /**
+     * Get a peer connection state given a peer Node Id, local Node Id and Peer's Encryption Key Id.
+     *
+     * @param nodeId is the connection to find (based on nodeId). Note that initial connections
+     *        do not have a node id set. Use this if you know the node id should be set.
+     * @param localNodeId The connection must correspond to the given local node ID.
+     * @param admins List of administrators that have commissioned this device.
+     * @param peerKeyId Encryption key ID used by the peer node.
+     * @param begin If a member of the pool, will start search from the next item. Can be nullptr to search from start.
+     *
+     * @return the state found, nullptr if not found
+     */
+    CHECK_RETURN_VALUE
+    PeerConnectionState * FindPeerConnectionState(Optional<NodeId> nodeId, NodeId localNodeId,
+                                                  Transport::AdminPairingTable<> * admins, uint16_t peerKeyId,
```

Comment:
What's the purpose of AdminPairingTable being templated on size if we can't use non-default sizes with src/transport?

Reply:
My thinking is that different devices may support different number of admins based on their processing power and memory constraints. Maybe there are better ways to achieve it.

Reply:
We were using config #defines for this.

If using a template, we must make methods that take an AdminPairingTable templates, too. If we just take AdminPairingTable<> then any non-default size will break the build with a type mismatch.

Reply:
Yeah, #define will work better. I'll update the PR.